docs/source/community/contributing.mdx

```diff
@@ -0,0 +1,116 @@
+specific language governing permissions and limitations under the License.
+-->
+
+# Adding support for new architectures
+
+
+With GitHub, I usually insert a blockquote.
+
+> **_NOTE:_** ❗This section does not apply to the decoder model’s inference with autoregressive sampling integrated through `transformers-neuronx`. If you want to add support for these models, please open an issue on the Optimum Neuron GitHub repo, and ping maintainers for help.
+
+You want to export and run a new model on AWS Inferentia or Trainium? Check the guideline, and submit a pull request to [🤗 Optimum Neuron's GitHub repo](https://github.com/huggingface/optimum-neuron/)! 
+
+To support a new model architecture in the Optimum Neuron library here are some steps to follow:
+
+1. Implement a custom Neuron configuration.
+2. Export and validate the model.
+3. Contribute to the GitHub repo. 
+
+## Implement a custom Neuron configuration
+
+To support the export of a new model to a Neuron compatible format, the first thing to do is to define a Neuron configuration, describing how to export the PyTorch model by specifying:
+
+1. The input names.
+2. The output names.
+3. The dummy inputs used to trace the model. (The Neuron Compiler asks this to record the computational graph and embed it to a TorchScript module.)
+4. The compilation arguments used to control the trade-off between hardware efficiency (latency, throughput) and accuracy.
+
+Depending on the choice of model and task, we represent the data above with configuration classes. Each configuration class is associated with
+a specific model architecture, and follows the naming convention `ArchitectureNameNeuronConfig`. For instance, the configuration that specifies the Neuron
+export of BERT models is `BertNeuronConfig`.
+
+Since many architectures share similar properties for their Neuron configuration, 🤗 Optimum adopts a 3-level class hierarchy:
+
+1. Abstract and generic base classes. These handle all the fundamental features, while being agnostic to the modality (text, image, audio, etc).
+2. Middle-end classes. These are aware of the modality. Multiple config classes could exist for the same modality, depending on the inputs they support. They specify which input generators should be used for generating the dummy inputs, but remain model-agnostic.
+3. Model-specific classes like the `BertNeuronConfig` mentioned above. These are the ones actually used to export models.
+
+### Example: Adding support for ESM models
+
+Here we take the support of [ESM models](https://huggingface.co/docs/transformers/model_doc/esm#esm) as an example. Let's create an `EsmNeuronConfig` class in the `optimum/exporters/neuron/model_configs.py`.
+
+When an Esm model interprets as a text encoder, we are able to inherit from the middle-end class [`TextEncoderNeuronConfig`](https://github.com/huggingface/optimum-neuron/blob/v0.0.18/optimum/exporters/neuron/config.py#L36). 
```

Comment:
Not sure about the phrasing here. You want mean "when an ESM model acts a text encoder" because it can act as something else?

Reply:
Theoretically, it could act [as a decoder as well according to the doc of ESM](https://huggingface.co/docs/transformers/model_doc/esm#:~:text=The%20model%20can%20behave%20as%20an%20encoder%20(with%20only%20self%2Dattention)%20as%20well%20as%20a%20decoder%2C). But I did not see seq2seq modeling yet implemented in transformers although there seems to be some placeholders for it.